src/Traits/ReflectionFunctionLikeTrait.php

```diff
@@ -339,6 +436,80 @@ public function isVariadic()
      */
     public function returnsReference()
     {
+        if (!$this->functionLikeNode) {
+            $this->initializeInternalReflection();
+            return parent::returnsReference();
+        }
         return $this->functionLikeNode->returnsByRef();
     }
+
+    private function getRefParam(BaseReflectionParameter $orig)
+    {
+        $nullableImplied = false;
+        $builder = new ParamNodeBuilder($orig->name);
+        if ($orig->isDefaultValueAvailable() || $orig->isOptional()) {
+            if ($orig->isDefaultValueAvailable()) {
+                if (method_exists($orig, 'isDefaultValueConstant') && $orig->isDefaultValueConstant()) {
+                    $constNameParts = explode('::', $orig->getDefaultValueConstantName(), 2);
```

Comment:
I'm fine with moving part of this to `NodeExpressionResolver`,  but it seems built to process nodes rather than strings. How would you recommend I delegate this to the parser? My gut reaction would be to give it something like:

```php
sprintf('<?php function foo($bar = %s) {}', $orig->getDefaultValueConstantName())
```

then pluck out the resulting AST nodes for the default value. Is there a cleaner way?